pkg/statedb2/metrics.go

```diff
@@ -0,0 +1,89 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package statedb2
+
+import (
+	"github.com/cilium/cilium/pkg/metrics"
+	"github.com/cilium/cilium/pkg/metrics/metric"
+)
+
+type Metrics struct {
+	// How long a read transaction was held.
+	WriteTxnDuration metric.Vec[metric.Observer]
+	// How long it took to acquire a write transaction for all tables.
+	WriteTxnAcquisition metric.Vec[metric.Observer]
+	// How long writers were blocked while waiting to acquire a write transaction for a specific table.
+	TableContention metric.Vec[metric.Gauge]
+	// The amount of objects in a given table.
+	TableObjectCount metric.Vec[metric.Gauge]
+	// The current revision of a given table.
+	TableRevision metric.Vec[metric.Gauge]
+	// The amount of delete trackers for a given table.
+	TableDeleteTrackerCount metric.Vec[metric.Gauge]
+	// The amount of objects in the graveyard for a given table.
+	TableGraveyardObjects metric.Vec[metric.Gauge]
+	// The lowest revision of a given table that has been processed by the graveyard garbage collector.
+	TableGraveyardLowWatermark metric.Vec[metric.Gauge]
+	// The time it took to clean the graveyard for a given table.
+	TableGraveyardCleaningDuration metric.Vec[metric.Observer]
+}
+
+func NewMetrics() Metrics {
+	return Metrics{
+		WriteTxnDuration: metric.NewHistogramVec(metric.HistogramOpts{
+			Namespace: metrics.CiliumAgentNamespace,
```

Comment:
The namespace should be configurable via `NewDB`... or via metrics registry?

Reply:
The registry should set the namespace of very collected metric.